numcompress/numcompress.py

```diff
@@ -81,3 +84,14 @@ def decompress_number(text, index):
             break
 
     return index, (~result >> 1) if (result & 1) != 0 else (result >> 1)
+
+
+def compress_ndarray(series, precision=3):
+    return f'{compress(list(series.shape), precision=0)}{SEPARATOR}{compress(series.flatten().tolist(), precision)}'
```

Comment:
We don't need to compress the dimension. Space savings are negligible. It's much useful for people to read/see - 
`10*10,k@ojcA~ka...` 
than
`B_pR?,k@ojcA~ka...`

Prior is just more human friendly - Quick for debugging, mental processing etc.

Reply:
updated